kolibri/core/tasks/exceptions.py

```diff
@@ -15,3 +15,14 @@ class JobNotFound(Exception):
 
 class JobNotRestartable(Exception):
     pass
+
+
+class ErrorSavedWithJob(Exception):
+    def __init__(self, message, prior_type):
+        if message is None or message == "":
+            message = "This job had an error before it was saved to the database."
+        self.prior_type = prior_type
+        super().__init__(message)
```

Comment:
For a Python 2 compatible super invocation, this needs to be called like this:
```suggestion
        super(ErrorSavedWithJob, self).__init__(message)
```